include/highfive/H5Easy.hpp

```diff
@@ -39,20 +39,151 @@
 
 namespace H5Easy {
 
+using HighFive::Attribute;
 using HighFive::AtomicType;
 using HighFive::Chunking;
 using HighFive::DataSet;
 using HighFive::DataSetCreateProps;
 using HighFive::DataSpace;
+using HighFive::Deflate;
 using HighFive::Exception;
 using HighFive::File;
 using HighFive::ObjectType;
+using HighFive::Shuffle;
 
 ///
 /// \brief Write mode for DataSets
 enum class DumpMode {
-    Create, /*!< Dump only if DataSet does not exist, otherwise throw. */
-    Overwrite /*!< If DataSet already exists overwrite it if data has the same shape, otherwise throw. */
+    Create = 0, /*!< Dump only if DataSet does not exist, otherwise throw. */
+    Overwrite = 1 /*!< Create or overwrite if DataSet of correct shape exists, otherwise throw. */
+};
+
+///
+/// \brief Enable/disable automatic flushing after write operations.
+enum class Flush
+{
+    False = 0, /*!< No automatic flushing. */
+    True = 1 /*!< Automatic flushing. */
+};
+
+///
+/// \brief Enable/disable compression for written DataSets.
+enum class Compression
+{
+    False = 0, /*!< No compression. */
+    Medium = 5, /*!< Medium compression (deflate level 5). */
+    High = 9 /*!< High compression (deflate level 9). */
+};
+
+///
+/// \brief Set compression-level for written DataSets.
+class CompressionLevel
+{
+public:
+    CompressionLevel(unsigned deflate_level);
+
+    inline unsigned get() const;
+
+private:
+    unsigned m_deflate_level;
```

Comment:
Compression or deflation? The naming here is inconsistent.

Reply:
Well.. If you look where it is used 
```cpp
props.add(Deflate(options.getDeflateLevel()));
```
deflate makes sense (for at least some of the settings). But at the same time for most users I would say that "compression" is more logical. So my preference is what I did. But it would be good to hear your preference.